src/shared/utils/calculate.ts

```diff
@@ -0,0 +1,36 @@
+import { CommentWithReComments } from "@detail/CommentBox";
+
+export const calculateNightsAndDays = (start: string, end: string) => {
+  const startDate = new Date(start);
+  const endDate = new Date(end);
+  const timeDiff = endDate.getTime() - startDate.getTime(); // 밀리초 단위로 변환
+  const nights = timeDiff / (1000 * 60 * 60 * 24); // 날짜 차이 계산
+  const days = nights + 1;
+  if (nights === 0) return "(당일)";
+  return `(${nights}박 ${days}일)`;
+};
+
+export const countCommentsAndReComments = (
+  comments: CommentWithReComments[],
+): number => {
+  // 댓글의 개수
+  const totalComments = comments?.length;
```

Comment:
[Lv1] `comments`가 무조건 들어와야 하는 상황이라면  
```suggestion
  const totalComments = comments.length;
```
optional 연산자가 없어도 될것같습니다!
만약 빈 배열이 아닌 다른 객체가 들어올 확률이 있다면 

```suggestion
  const totalComments = comments?.length ?? 0;
```
위와같이 예외상황에서의 기본 값을 같이 넣어주는게 좋을것같습니다!